doc/dev/api/server/Ppx_syntax.wiki

```diff
@@ -0,0 +1,67 @@
+=RPC ppx syntax extension
+
+The ocsigen-start syntax extension allows to write, in one function, both the
+client and server part of your RPC.
+
+
+==Sections
+
+The location of code execution is specified by section annotations. 3 types
+of constructions are available to perform RPC sections:
+
+<<code language="ocaml"|
+[%%rpc
+  let r = ...
+]
+
+>>
+
+Construction to compute an RPC using the {{connected_wrapper}} API.
+
+<<code language="ocaml"|
+[%%crpc
+  let r = ...
+]
+
+[%%crpc_opt
+  let c = ...
+]
+>>
+
+Construction to compute an RPC using the {{connected_rpc}} API.
+
+
+Constructions have a shortcut:
+
+<<code language="ocaml"|
+let%rpc r = ...
+
+let%crpc c = ...
+>>
+
+
+Sections are only available at top level and cannot be nested.
+
+In {{{.eliomi}}} files, an RPC is declared as shared function.
+
+==Restrictions
+
+As the RPC will be computed on both client-side and server-side,
```

Comment:
This is not exactly the reason.
The reason is that you have to explicite a monomorphic type to be able to translate it to json, and to communicate between server and client (on this part you are right).